cmd/kube-proxy/app/server.go

```diff
@@ -658,9 +674,13 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
 
 // CleanupAndExit remove iptables rules and exit if success return nil
 func (s *ProxyServer) CleanupAndExit() error {
-	encounteredError := userspace.CleanupLeftovers(s.IptInterface)
-	encounteredError = iptables.CleanupLeftovers(s.IptInterface) || encounteredError
-	encounteredError = ipvs.CleanupLeftovers(s.IpvsInterface, s.IptInterface, s.IpsetInterface, s.CleanupIPVS) || encounteredError
+	var encounteredError bool
+	for _, iface := range []*utiliptables.Interface{s.Ipt4Interface, s.Ipt6Interface} {
+		encounteredError := userspace.CleanupLeftovers(*iface)
+		encounteredError = iptables.CleanupLeftovers(*iface) || encounteredError
+		encounteredError = ipvs.CleanupLeftovers(s.IpvsInterface, *iface, s.IpsetInterface, s.CleanupIPVS) || encounteredError
+	}
+
 	if encounteredError {
```

Comment:
should the if go inside the for loop or do we have to wait until we clean all the interfaces?